sympy/series/fourier.py

```diff
@@ -440,16 +438,213 @@ def __sub__(self, other):
         return self.__add__(-other)
 
 
-class FiniteFourierSeries(Basic):
-    def __new__(cls, *args):
-        obj = Basic.__new__(cls, *args)
-        return obj
+class FiniteFourierSeries(FourierSeries):
+    r"""Represents Finite Fourier sine/cosine series.
 
-    def truncate(self, n=3):
-        return Add(*self._args)
+       For how to compute Fourier series, see the :func:`fourier_series`
+       docstring.
+
+       Parameters
+       ==========
+       f : Expr
+            Expression for finding fourier_series
+
+       limits : ( x, start, stop)
+            x is the independent variable for the expression f
+            (start, stop) is the period of the fourier series
+
+       exprs: (a0, an, bn) or Expr
+            a0 is the constant term a0 of the fourier series
+            an is a dictionary of coefficients of cos terms
+             `an[k] = coefficient of cos(pi*(k/L)*x)`
+            bn is a dictionary of coefficients of sin terms
+             `bn[k] = coefficient of sin(pi*(k/L)*x)`
+
+            or exprs can be an expression to be converted to fourier form
+
+       Methods
+       =======
+       This class is an extension of FourierSeries class.
+       Please refer to sympy.series.fourier.FourierSeries for
+       further information.
+
+       See Also
+       ========
+
+       sympy.series.fourier.FourierSeries
+       sympy.series.fourier.fourier_series
+       """
+
+    def __new__(cls, f, limits, exprs):
+
+        if not (type(exprs) == tuple and len(exprs) == 3):  # exprs is not of form (a0, an, bn)
+
+            # Converts the expression to fourier form
+            exprs = exprs.as_coeff_add()
+            exprs = exprs[0] + Add(*[TR10(i) for i in exprs[1]])
+            exp_ls = exprs.expand(trig=False, power_base=False, power_exp=False, log=False).as_coeff_add()
+
+            a0 = exp_ls[0]
+            x = limits[0]
+            L = abs(limits[2] - limits[1]) / 2
+
+            a = Wild('a', properties=[lambda k: k.is_Integer, lambda k: k != S.Zero, ])
```

Comment:
There could be `k is not S.Zero` as `S.Zero` is a singleton. It will save a few function calls.

Reply:
I don't know but k can be `0` at some time. But `S.Zero == 0` while `S.Zero is not 0`

Reply:
SymPy 0 is `S.Zero`. Can `k` be a non-Sympy number?

Reply:
I guess you are right. Since anything reaching to that part of code has been sympified, `0` would eventually be `S.Zero`. Should I change it? This would initiate another build for tests. If there is anything more you want to suggest I would add it in this commit only to save some build time.

Reply:
I think you could change it. `S.Zero` was designed to make quick tests for equality possible. Maybe you could wait for some time after a new commit. It would be possible to push several commits at a time. I have not had time to finish reviewing yet.